deltacat/storage/rivulet/dataset.py

```diff
@@ -53,6 +55,50 @@ def from_pydict(cls, base_uri: str, data: Dict[str, List[Any]], primary_key: str
         riv_schema = Schema.from_pyarrow_schema(table.schema, primary_key)
         return cls(base_uri, field_groups=[PydictFieldGroup(data, riv_schema)])
 
+    @classmethod
+    def from_parquet(
+        cls,
+        file_uri: str,
+        primary_key: str,
+        base_uri: Optional[str] = None,
+        schema_mode: str = "union",
+    ):
+        """
+        Create a Dataset from parquet files.
+
+        Args:
+            base_uri: Base URI for the dataset, where dataset metadata is stored. If not specified, will be placed in ${file_uri}/riv-meta
+            file_uri: Path to parquet file(s)
+            primary_key: Field to use as primary key
+            schema_mode: Schema combination mode. Options:
+                - 'union': Use unified schema with all columns
+                - 'intersect': Use only common columns across files
+
+        Returns:
+            Dataset: New dataset instance with the schema automatically inferred from the source parquet files
+        """
+        base_uri = base_uri or os.path.join(file_uri, "riv-meta")
```

Comment:
`os.path.join` will produce incorrect file paths due to inclusion of '\' if run on Windows and the URI refers to, say, an S3 object. We may either want to replace this with `posixpath.join` or simply `"/".join` (together with ensuring that all input paths are normalized).